packages/core/src/actions/transactions/fetchTransaction.ts

```diff
@@ -0,0 +1,32 @@
+import { providers } from 'ethers'
+
+import { Hash } from '../../types'
+import { getProvider } from '../providers'
+
+export type FetchTransactionArgs = {
+  /** Chain ID used to validate if the signer is connected to the target chain */
+  chainId?: number
+  /** Transaction hash */
+  hash: Hash
+}
+
+export type FetchTransactionResult = providers.TransactionResponse
```

Comment:
Returns unmodified [`TransactionResponse`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse) here, but maybe we start constructing our own objects instead of returning ethers responses directly?

[`TransactionReceipt`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt) also has some useful properties in it (e.g. `logs`).